src/chemistry/aerosol/cloud_aqueous_chemistry.F90

```diff
@@ -0,0 +1,894 @@
+          !-----------------------------------------------------------------
+          px = hehno3(i,k) * Ra * tz * xl
+          hno3g(i,k) = (xhno3(i,k)+xno3(i,k))/(1._r8 + px)
+
+          !------------------------------------------------------------------------
+          !        ... h2o2
+          !------------------------------------------------------------------------
+          px = heh2o2(i,k) * Ra * tz * xl
+          h2o2g =  xh2o2(i,k)/(1._r8+ px)
+
+          !------------------------------------------------------------------------
+          !         ... so2
+          !------------------------------------------------------------------------
+          px = heso2(i,k) * Ra * tz * xl
+          so2g =  xso2(i,k)/(1._r8+ px)
+
+          !------------------------------------------------------------------------
+          !         ... o3
+          !------------------------------------------------------------------------
+          px = heo3(i,k) * Ra * tz * xl
+          o3g =  xo3(i,k)/(1._r8+ px)
+
+          !------------------------------------------------------------------------
+          !         ... nh3
+          !------------------------------------------------------------------------
+          px = henh3(i,k) * Ra * tz * xl
+          if (id_nh3>0) then
+             nh3g(i,k) = (xnh3(i,k)+xnh4(i,k))/(1._r8+ px)
+          else
+             nh3g(i,k) = 0._r8
+          endif
+
+          !-----------------------------------------------
+          !       ... Aqueous phase reaction rates
+          !           SO2 + H2O2 -> SO4
+          !           SO2 + O3   -> SO4
+          !-----------------------------------------------
+
+          !------------------------------------------------------------------------
+          !       ... S(IV) (HSO3) + H2O2
+          !------------------------------------------------------------------------
+          rah2o2 = 8.e4_r8 * EXP( -3650._r8*work1(i) )  &
+               / (.1_r8 + xph(i,k))
+
+          !------------------------------------------------------------------------
+          !        ... S(IV)+ O3
+          !------------------------------------------------------------------------
+          rao3   = 4.39e11_r8 * EXP(-4131._r8/tz)  &
+               + 2.56e3_r8  * EXP(-996._r8 /tz) /xph(i,k)
+
+          !-----------------------------------------------------------------
+          !       ... Prediction after aqueous phase
+          !       so4
+          !       When Cloud is present
+          !
+          !       S(IV) + H2O2 = S(VI)
+          !       S(IV) + O3   = S(VI)
+          !
+          !       reference:
+          !           (1) Seinfeld
+          !           (2) Benkovitz
+          !-----------------------------------------------------------------
+
+          !............................
+          !       S(IV) + H2O2 = S(VI)
+          !............................
+
+          IF (XL .ge. 1.e-8_r8) THEN    !! WHEN CLOUD IS PRESENTED
```

Comment:
presented what? why is `1e-8` the value we care about? Is it empirical? Numerical? If numerical, was it derived when we only had single precision and we could make it smaller now?

Reply:
Yea it's a small positive definite. You can use greater than zero instead of great or equal to a positive definite, I suppose. There are a lot f these small numbers in different parts of CAM-Chem. There are 1e-20, 1e-30, and so on.